pypowerwall/cloud/pypowerwall_cloud.py

```diff
@@ -0,0 +1,997 @@
+        config = self.get_site_config()
+        if config is None:
+            data = None
+        else:
+            default_real_mode = lookup(config, ("response", "default_real_mode"))
+            backup_reserve_percent = lookup(config, ("response", "backup_reserve_percent")) or 0
+            # backup_reserve_percent is scaled to keep 5% buffer at bottom
+            backup = (backup_reserve_percent + (5 / 0.95)) * 0.95
+            data = {
+                "real_mode": default_real_mode,
+                "backup_reserve_percent": backup
+            }
+        return data
+
+    def get_api_system_status(self) -> Optional[Union[dict, list, str, bytes]]:
+        power = self.get_site_power()
+        config = self.get_site_config()
+        battery = self.get_battery()
+        if power is None or config is None or battery is None:
+            data = None
+        else:
+            # timestamp = lookup(power, ("response", "timestamp"))
+            solar_power = lookup(power, ("response", "solar_power"))
+            # battery_power = lookup(power, ("response", "battery_power"))
+            # load_power = lookup(power, ("response", "load_power"))
+            grid_services_power = lookup(power, ("response", "grid_services_power"))
+            # grid_status = lookup(power, ("response", "grid_status"))
+            # grid_services_active = lookup(power, ("response", "grid_services_active"))
+            battery_count = lookup(config, ("response", "battery_count"))
+            total_pack_energy = lookup(battery, ("response", "total_pack_energy"))
+            energy_left = lookup(battery, ("response", "energy_left"))
+            nameplate_power = lookup(config, ("response", "nameplate_power"))
+            # nameplate_energy = lookup(config, ("response", "nameplate_energy"))
+            if lookup(power, ("response", "island_status")) == "on_grid":
+                grid_status = "SystemGridConnected"
+            else:  # off_grid or off_grid_unintentional
+                grid_status = "SystemIslandedActive"
+                # "grid_status": "Active"
+                if lookup(power, ("response", "grid_status")) == "Active":
+                    grid_status = "SystemGridConnected"
+            data = API_SYSTEM_STATUS_STUB  # TODO: see inside API_SYSTEM_STATUS_STUB definition
+            data.update({
+                "nominal_full_pack_energy": total_pack_energy,
+                "nominal_energy_remaining": energy_left,
+                "max_charge_power": nameplate_power,
+                "max_discharge_power": nameplate_power,
+                "max_apparent_power": nameplate_power,
+                "grid_services_power": grid_services_power,
+                "system_island_state": grid_status,
+                "available_blocks": battery_count,
+                "solar_real_power_limit": solar_power,
+                "blocks_controlled": battery_count,
+            })
+
+        return data
+
+    @not_implemented_mock_data
```

Comment:
Added a decorator here that warns users that fake/mock data is returned

Reply:
Great idea!